### PR TITLE
Add mvnbin build pattern

### PR DESCRIPTION
**Pull request by AthenasJimenez**
This pattern would add the ability to create bootstraps specfiles for
maven dependencies(aka artifacts). These boostraps would be use as a
way to solve the dependencies for Java projects that use Maven as a
build tool.

Signed-off-by: Athenas Jimenez <athenas.jimenez.gonzalez@intel.com>

### CODE REVIEW

File: autospec/tarball.py [hunk 1]
```diff
@@ -501,6 +501,8 @@ def find_extract(tar_path, tarfile):
         extract_cmd, tar_prefix = build_unzip(tar_path)
     elif tarfile.lower().endswith('.gem'):
         extract_cmd, tar_prefix = build_gem_unpack(tar_path)
+    elif tarfile.lower().endswith('.jar'):
```
Comment (bryteise) [inline autospec/tarball.py]: Is this handling alternate to maven java build systems case?

File: autospec/specfiles.py [hunk 2]
```diff
@@ -1495,8 +1528,8 @@ def write_meson_pattern(self):
 
         if config.config_opts['use_avx2']:
             self._write_strip('CFLAGS="$CFLAGS -m64 -march=haswell" CXXFLAGS="$CXXFLAGS -m64 -march=haswell " LDFLAGS="$LDFLAGS -m64 -march=haswell" '
-                              'meson --prefix /usr --libdir=/usr/lib64/haswell --buildtype=plain {0} '
-                              '{1} builddiravx2'.format(config.extra_configure, config.extra_configure64))
+                              'meson --prefix /usr --libdir=/usr/lib64/haswell --buildtype=plain'
+                              '{0} {1} builddiravx2'.format(config.extra_configure, config.extra_configure64))
```
Comment (bryteise) [inline autospec/specfiles.py]: This shouldn't be in this change, please rebase.

**Comment by bryteise** [comment]
Looks pretty good to me other than the extra bit that needs to be rebased out.

File: autospec/tarball.py (hunk 1 above)
Comment (AthenasJimenez) [reply autospec/tarball.py]: No, this is needed by the maven bin case to get the License info and description.

File: autospec/specfiles.py (hunk 2 above)
Comment (AthenasJimenez) [reply autospec/specfiles.py]: O gosh, forgot to check for new changes. I'll fix it.

**Comment by bryteise** [comment]
Great, thanks!

**Pull request merged**